app/graphql/types/invoice_subscription/object.rb

```diff
@@ -16,8 +16,38 @@ class Object < Types::BaseObject
 
       field :charges_from_datetime, GraphQL::Types::ISO8601DateTime, null: true
       field :charges_to_datetime, GraphQL::Types::ISO8601DateTime, null: true
+
+      field :in_advance_charges_from_datetime, GraphQL::Types::ISO8601DateTime, null: true
+      field :in_advance_charges_to_datetime, GraphQL::Types::ISO8601DateTime, null: true
+
       field :from_datetime, GraphQL::Types::ISO8601DateTime, null: true
       field :to_datetime, GraphQL::Types::ISO8601DateTime, null: true
+
+      def in_advance_charges_from_datetime
+        return nil unless should_use_in_advance_charges_interval
+
+        charge_pay_in_advance_interval[:charges_from_date]
+      end
+
+      def in_advance_charges_to_datetime
+        return nil unless should_use_in_advance_charges_interval
+
+        charge_pay_in_advance_interval[:charges_to_date]
+      end
+
+      def should_use_in_advance_charges_interval
+        return @should_use_in_advance_charges_interval if defined? @should_use_in_advance_charges_interval
+
+        @should_use_in_advance_charges_interval =
+          object.fees.charge_kind.any? &&
+            object.subscription.plan.charges.where(pay_in_advance: true).any? &&
```

Comment:
Layout/MultilineOperationIndentation: Align the operands of an expression in an assignment spanning multiple lines.